compat/maven-settings-builder/src/main/java/org/apache/maven/settings/building/DefaultSettingsBuilder.java

```diff
@@ -251,4 +255,29 @@ public Object execute(String expression, Object value) {
 
         return result;
     }
+
+    private List<Server> serversByIds(List<Server> servers) {
+
+        if (servers.stream().allMatch(server -> server.getIds().isEmpty())) {
+            return servers;
+        }
+
+        return servers.stream()
+                .flatMap(server -> {
+                    List<String> ids = server.getIds();
+                    if (ids.isEmpty()) {
+                        return Stream.of(server);
+                    } else {
+                        return Stream.concat(Stream.of(server), ids.stream().map(id -> newServer(server, id)));
+                    }
+                })
+                .collect(Collectors.toList());
+    }
+
+    private Server newServer(Server server, String id) {
+        return new Server(org.apache.maven.api.settings.Server.newBuilder(server.getDelegate(), false)
+                .id(id)
+                .ids(Collections.emptyList())
```

Comment:
```suggestion
                .ids(List.of())
```

Reply:
dont do it here makes no sense, it will never be done this way.

- https://github.com/apache/maven/pull/2322

Reply:
Don't do _what_ here ?

Reply:
refactoring for no reason, this can only be done by CPU.

Reply:
Well, I you saw, my manual refactoring was much better than your CPU one.  I still have an advantage here ;-)

And this is not refactoring, since this is new code !

Reply:
> I still have an advantage here ;-)

you never finish. I make suppression and its done.

Reply:
> new code

technically yes, so its good to make things right from beginning. Still need for automation.

Reply:
https://github.com/openrewrite/rewrite-static-analysis/pull/558